Sources/Task/TaskRecover.swift

```diff
@@ -47,4 +47,18 @@ extension Task {
         return Task<SuccessValue>(future: future, cancellation: cancel)
         #endif
     }
+
+    public func recoverWithNewTask(upon executor: Executor, start startNextTask: @escaping(Error) -> Task<SuccessValue>) -> Task<SuccessValue> {
```

Comment:
Agreed that the naming is not ideal here. 

I really like `fallback(upon:to:)` so if I don't see any other suggestions I'd change it for this option.